apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/AppServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package com.ctrip.framework.apollo.portal.service;
+
+import com.ctrip.framework.apollo.common.entity.App;
+import com.ctrip.framework.apollo.common.entity.AppNamespace;
+import com.ctrip.framework.apollo.openapi.entity.ConsumerRole;
+import com.ctrip.framework.apollo.openapi.repository.ConsumerRoleRepository;
+import com.ctrip.framework.apollo.portal.PortalApplication;
+import com.ctrip.framework.apollo.portal.entity.po.Favorite;
+import com.ctrip.framework.apollo.portal.entity.po.Permission;
+import com.ctrip.framework.apollo.portal.entity.po.RolePermission;
+import com.ctrip.framework.apollo.portal.entity.po.UserRole;
+import com.ctrip.framework.apollo.portal.repository.AppNamespaceRepository;
+import com.ctrip.framework.apollo.portal.repository.AppRepository;
+import com.ctrip.framework.apollo.portal.repository.FavoriteRepository;
+import com.ctrip.framework.apollo.portal.repository.PermissionRepository;
+import com.ctrip.framework.apollo.portal.repository.RolePermissionRepository;
+import com.ctrip.framework.apollo.portal.repository.RoleRepository;
+import com.ctrip.framework.apollo.portal.repository.UserRoleRepository;
+
+import org.apache.commons.lang.math.RandomUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import java.util.List;
+import java.util.Objects;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {PortalApplication.class})
+public class AppServiceTest {
```

Comment:
This test is kind of simple...we may need a more detailed test like [RolePermissionServiceTest](https://github.com/ctripcorp/apollo/blob/master/apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/spi/defaultImpl/RolePermissionServiceTest.java). We could use direct sqls to prepare different kinds of data